### ISSUE TITLE
Cross compile to target only

### ISSUE THREAD

**Issue by gilesknap** [epics-containers/ioc-pmac#26]
This generic IOC is supposed to only compile for the target when crosscompiling for RTEMS-beatnik.

I believe I had this working at some point but it appears not to be at the moment.

For this to work each support module should have `VALID_BUILDS=Host` in `configure/CONFIG_SITE.Common.linux-x86_64`.

This is supposed to happen in `ibek-support/_global/install.sh`.

**Comment by gilesknap** [comment]
Current state:

- `ibek-support/_global/install.sh` no longer exists. `VALID_BUILDS=Host` is now written by ansible in two places, both gated on `is_rtems` ([`group_vars/all.yml:36`](https://github.com/epics-containers/ibek-support/blob/5cb8932434d44858efabc6e9f953e9581dd79786/_ansible/group_vars/all.yml#L36), `'RTEMS' in target_arch`):
  - per support module: [`_ansible/roles/support/tasks/pre_build.yml:17-24`](https://github.com/epics-containers/ibek-support/blob/5cb8932434d44858efabc6e9f953e9581dd79786/_ansible/roles/support/tasks/pre_build.yml#L17-L24), into `configure/CONFIG_SITE.Common.linux-x86_64` (`config_linux_target`, [`group_vars/all.yml:29`](https://github.com/epics-containers/ibek-support/blob/5cb8932434d44858efabc6e9f953e9581dd79786/_ansible/group_vars/all.yml#L29))
  - for the IOC: [`_ansible/roles/ioc/tasks/main.yml:13-20`](https://github.com/epics-containers/ibek-support/blob/5cb8932434d44858efabc6e9f953e9581dd79786/_ansible/roles/ioc/tasks/main.yml#L13-L20)

  So the mechanism this issue asks for is in place again.
- It cannot be checked right now, because ioc-pmac does not build for RTEMS at all. The `RTEMS-beatnik` matrix entry is commented out in [`.github/workflows/build.yml:21-25`](https://github.com/epics-containers/ioc-pmac/blob/141f2f8/.github/workflows/build.yml#L21-L25) (the RTEMS WAF does not work with Python 3.12). The Dockerfile bases on `ioc-asyn${IMAGE_EXT}-developer:4.46`, and there is no public `ioc-asyn-rtems-beatnik-developer` image on ghcr. The newest `epics-base-rtems-beatnik-developer` tag is `7.0.9ec4b1`/`7.0.9rtems6.1ec1`, with no `7.0.9ec5`. The last `ioc-pmac-rtems-beatnik-developer` image is `2025.1.1b1`.

This issue is really blocked on getting an RTEMS build of ioc-pmac working again (RTEMS 6 epics-base, then an RTEMS ioc-asyn or building asyn in ioc-pmac). Once that happens, the ansible `VALID_BUILDS` tasks above are what to verify. I suggest either closing this in favour of a "restore RTEMS-beatnik build of ioc-pmac" issue, or retitling it.

_Triage comment drafted by Claude Code._
